code/__HELPERS/icons.dm

```diff
@@ -872,6 +872,52 @@ The _flatIcons list is a cache for generated icon files.
 	del(atom_icon)
 	return text_image
 
+// Pick a random animal instead of the icon, and use that instead
+/proc/getRandomAnimalImage(atom/A)
+	var/list/icons = list()
+	var/icon/selected
+
+	icons += new/icon('icons/mob/animal.dmi', "cow")

```

Comment:
@RemieRichards Remie, why, making mob icons out of friendly gold slime spawns is turning out to be requiring a lot of extra work to make work, especially with chickens because of how they only pick valid icon_state in their New(), and rabbits are also a child class of rabbits, and IT'S SO MUCH WORK. We could make chickens have an initial valid icon_state, but then we lose the ability to have different coloured chickens in filters, same with rabbits, and I've already special cased butterflies, and wew.